README.md

```diff
@@ -4,252 +4,83 @@
   <a href="https://github.com/actions/setup-java"><img alt="GitHub Actions status" src="https://github.com/actions/setup-java/workflows/Main%20workflow/badge.svg"></a>
 </p>
 
-This action sets up a java environment for use in actions by:
-
-- optionally downloading and caching a requested version of java by version and adding to PATH. Default downloads are populated from the [Zulu Community distribution of OpenJDK](http://static.azul.com/zulu/bin/)
-- registering problem matchers for error output
-
-# Usage
-
-See [action.yml](action.yml)
-
-## Basic
+This action provides the following functionality for GitHub Actions runners:
+- Downloading and setting up a requested version of Java. See [Usage](#Usage) section for the list of supported distributions
+- Extracting and caching custom version of Java from local file
+- Configuring runner for publishing using Apache Maven
+- Configuring runner for publishing using Gradle
+- Configuring runner for using GPG private key
+- Registering problem matchers for error output
+
+## V2 vs V1
+- V2 has support of custom distributions and provides support of Zulu OpenJDK and Adoptium (former AdoptOpenJDK) out of the box. V1 supports only Zulu OpenJDK
+- V2 requires you to specify distribution along with the version. V1 defaults to Zulu OpenJDK, only version input is required. Follow [the migration guide](docs/switching-to-v2.md) to switch from V1 to V2
+
+## Usage
+Input `distribution` is mandatory. See [Supported distributions](../README.md#Supported-distributions) section for the list of available options.
+
+### Basic
 ```yaml
 steps:
 - uses: actions/checkout@v2
-- uses: actions/setup-java@v1
+- uses: actions/setup-java@v2-preview
   with:
-    java-version: '9.0.4' # The JDK version to make available on the path.
-    java-package: jdk # (jre, jdk, or jdk+fx) - defaults to jdk
-    architecture: x64 # (x64 or x86) - defaults to x64
+    distribution: 'adoptium' # See 'Supported distributions' for available options
+    java-version: '11'
 - run: java -cp java HelloWorldApp
 ```
-Examples of version specifications that the java-version parameter will accept:
-
-- A major Java version
-
-  e.g. ```6, 7, 8, 9, 10, 11, 12, 13, ...```
-
-- A semver Java version specification
-
-  e.g. ```8.0.232, 7.0.181, 11.0.4```
-
-  e.g. ```8.0.x, >11.0.3, >=13.0.1, <8.0.212```
 
-- An early access (EA) Java version
+#### Supported version syntax
+Input `java-version` supports version range or exact version in [SemVer](https://semver.org/) format:
+- major versions: `8`, `11`, `15`, `11.x`
+- more specific versions: `8.0.232`, `11.0.4`, `11.0`, `11.0.x`
+- an early access (EA) versions: `15-ea`, `15.0.0-ea`, `15.0.0-ea.2`
```

Comment:
`The "java-verion" input supports an exact version or a version range using [SemVer](https://semver.org/) notation:`

the last line I would change to just `early access...`